kubernetes/nodecollection.go

```diff
@@ -348,34 +348,59 @@ func ensureNodeSource(ctx context.Context, config KubeAgentConfig) (KubeAgentCon
 
 	firstNode := &nodes[0]
 
-	ip, port, err := clientSetNodeSource.NodeAddress(firstNode)
-	if err != nil {
-		return config, fmt.Errorf("error retrieving node addresses: %s", err)
-	}
-
-	if allowDirectConnect(config, nodes) {
-		// test node direct connectivity
-		d := directNodeEndpoints(ip, port)
-		success, err := checkEndpointConnections(config, &nodeHTTPClient, Direct, d.statsSummary())
+	directNodes := 0
+	proxyNodes := 0
+	failedDirect := 0
+	failedProxy := 0
+	directAllowed := allowDirectConnect(config, nodes)
+	for i := range nodes {
+		directlyConnected := false
+		ip, port, err := clientSetNodeSource.NodeAddress(&nodes[i])
 		if err != nil {
-			return config, err
+			return config, fmt.Errorf("error retrieving node addresses: %s", err)
 		}
-		if success {
-			return config, nil
+		if directAllowed {
+			// test node direct connectivity
+			d := directNodeEndpoints(ip, port)
+			success, err := checkEndpointConnections(config, &nodeHTTPClient, Direct, d.statsSummary())
+			if err != nil {
+				log.Warnf("Failed to connect to node [%s] directly with cause [%s]",
+					d.statsSummary(), err.Error())
+				failedDirect++
+			}
+			if success {
+				directlyConnected = true
+				directNodes++
+			}
+		}
+		if !directlyConnected {
```

Comment:
Does the agent attempt to query through proxy if unable to connect directly? I think we should stick with whatever connection option was set in the config

Reply:
The previous pattern was:
- Check if we are allowed to connect directly
- Attempt direct if so
- If not allowed to connect directly, or failed to connect directly, attempt proxy

This should follow the same pattern